hacep/src/main/java/org/kie/hacep/consumer/DroolsConsumerHandler.java

```diff
@@ -44,45 +45,85 @@
     private static final Logger logger = LoggerFactory.getLogger(DroolsConsumerHandler.class);
     private Logger loggerForTest;
     private Producer producer;
-    private DefaultSessionSnapShooter snapshooter;
-    private EnvConfig config;
+    private DefaultSessionSnapShooter sessionSnapShooter;
+    private EnvConfig envConfig;
     private KieSessionContext kieSessionContext;
     private CommandHandler commandHandler;
-    private SnapshotInfos infos;
+    private SnapshotInfos snapshotInfos;
     private boolean shutdown;
 
     public DroolsConsumerHandler(Producer producer, EnvConfig envConfig) {
```

Comment:
Please verify `KieSessionContext` field lifecycle, I'm quite sure it is initialized only in some code path and not in others:
-> initializeKieSessionContext -> initializeSessionContextWithSnapshotCheck -> initializeSessionContextFromSnapshot -> NPE?
-> initializeKieSessionContext -> initializeSessionContextWithSnapshotCheck -> createAndInitializeSessionContextWithoutSnapshot -> It seems OK
-> initializeKieSessionContext -> createAndInitializeSessionContextWithoutSnapshot -> It seems OK

Reply:
There was a init without the new before the call, fixed